core/clients/key_flow.go

```diff
@@ -131,6 +123,30 @@ func (c *KeyFlow) Init(cfg *KeyFlowConfig) error {
 	return c.validate()
 }
 
+// SetToken can be used to set an access and refresh token manually in the client
+// the other fields in the token field are determined by inspecting the token or setting default values
+func (c *KeyFlow) SetToken(accessToken, refreshToken string) error {
+	// We can safely use ParseUnverified because we are not authenticating the user,
+	// We are parsing the token just to get the expiration time claim
+	parsedAccessToken, _, err := jwt.NewParser().ParseUnverified(accessToken, &jwt.RegisteredClaims{})
+	if err != nil {
+		return fmt.Errorf("parse access token to read expiration time: %w", err)
+	}
+	exp, err := parsedAccessToken.Claims.GetExpirationTime()
+	if err != nil {
+		return fmt.Errorf("get expiration time from access token: %w", err)
+	}
+
+	c.token = &TokenResponseBody{
+		AccessToken:  accessToken,
+		ExpiresIn:    int(exp.Time.Unix()),
```

Comment:
Not related to this PR, but would it make sense to change this field name to `ExpiresAt`? As it's expiration time and not a countdown to it

Reply:
The field name is `ExpiresIn` to match the field name on the response from the ServiceAccounts API `/token` endpoint. Check https://docs.api.eu01.stackit.cloud/documentation/service-account/version/v2#tag/Public/operation/post-token

Reply:
I agree that it should be `ExpiresAt`, but that would have to be renamed in the API itself